transport/rest/taker.go

```diff
@@ -93,19 +93,23 @@ func (h *Handler) nameFromAddress(name, address string) (string, error) {
 
 func (h *Handler) resolveQuoteTokenNames(req *QuoteReq) error {
 	// has address but no name
-	InName, err := h.nameFromAddress(req.InToken, req.InTokenAddress)
-	if err != nil {
-		return err
-	}
-	if len(InName) > 0 {
-		req.InToken = InName
-	}
-	OutName, err := h.nameFromAddress(req.OutToken, req.OutTokenAddress)
-	if err != nil {
-		return err
+	if req.InToken == "" {
+		InName, err := h.nameFromAddress(req.InToken, req.InTokenAddress)
+		if err != nil {
+			return err
+		}
+		if len(InName) > 0 {
+			req.InToken = InName
```

Comment:
So `InName` is a slice / array ? Isn't this assingment wrong then? You want InToken to be `BSC`, not `['BSC']`.
Also, is it possible that `InName` will return more than 1 elements?

Reply:
mmm nope. inName is a string, not a slice. internally we work with stokenNames, but deddy may call us with either a name or an addrsss. if he comes with and adress and the inToken name eq "" empty string, we resolve that, hence the assignment